atomic_reactor/util.py

```diff
@@ -912,6 +914,26 @@ def _config_by_digest(self, image: ImageName, config_digest: str) -> dict:
         blob_config = config_response.json()
         return blob_config
 
+    def _config_and_id_from_manifest_list(
+        self, image: ImageName, manifest_list: dict, arch: Optional[str]
+    ) -> Tuple[dict, str]:
+        # if an architecture was specified, filter the correct architecture
+        filtered_manifests = (
+            m for m in manifest_list["manifests"]
+            if not arch or m.get("platform", {}).get("architecture") == arch
+        )
+        manifest = next(filtered_manifests, None)
```

Comment:
Logically, if arch is None or `''`, the `filtered_manifests` might have more than one manifests. If this happens, why only handle the first one?

Reply:
That would mean the caller doesn't care what architecture is being inspected, so we just inspect the first image in the list. Often it is not important to inspect a specific architecture, because we expect some properties to be the same between arches.

Btw, there may be multiple manifests even if arch is not falsy, e.g. `skopeo inspect --raw docker://docker.io/library/busybox:1.34.1 | jq '.manifests[] | select(.platform.architecture == "arm")'`
```json
{
  "digest": "sha256:0252da5f2df7425dcf48afb4bc337966dfeb2d87079ea3f7fe25051d5b9e9c26",
  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
  "platform": {
    "architecture": "arm",
    "os": "linux",
    "variant": "v5"
  },
  "size": 527
}
{
  "digest": "sha256:b27cc98025245c0e746b201d5c773faff99869ae58585090182e18d7c5e8a5e2",
  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
  "platform": {
    "architecture": "arm",
    "os": "linux",
    "variant": "v6"
  },
  "size": 527
}
{
  "digest": "sha256:7c416c3c45da3176e6160c81f2ed7b94265d98ebfeac11b12eb3ce3cc289644c",
  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
  "platform": {
    "architecture": "arm",
    "os": "linux",
    "variant": "v7"
  },
  "size": 527
}
```
I don't actually know if that's OK, but I hope it is, because OSBS has no concept of architecture variants